handlers/authentication.go

```diff
@@ -285,9 +279,37 @@ func performOpenIdAuthentication(w http.ResponseWriter, r *http.Request) bool {
 		return false
 	}
 
-	// Create business layer using the received id_token
-	expiresOn := time.Now().Add(time.Second * time.Duration(expiresInNumber))
+	// Set a default value for expiration date
+	expiresOn := time.Now().Add(time.Second * time.Duration(defaultSessionDuration))
+
+	// We check for the expiration date from the claim
+	if expClaim := idTokenClaims["exp"].(string); expClaim != "" {
+		expiresInNumber, err := strconv.Atoi(expClaim)
```

Comment:
Can you double-check this?
AFAIK, the `exp` claim is a Unix timestamp. If this is true, you should take the value as is (as the expiration time), rather than using it as a duration.

Reply:
Should be correct now.